server-core/engine/src/main/java/io/infectnet/server/engine/core/world/strategy/AStarPathFinderStrategy.java

```diff
@@ -0,0 +1,298 @@
+     * @return the new position nearby the original target
+     */
+    private Position resetTarget() {
+        Position newTarget =  targetPos;
+        Node targetNode = new Node(targetPos);
+
+        for(Node node : targetNode.getNeighbours()){
+            if(world.isPositionValidTile(node.getPosition())
+                    && !world.getTileByPosition(node.getPosition()).isBlockedOrOccupied()){
+                newTarget = node.getPosition();
+                break;
+            }
+        }
+        return newTarget;
+    }
+
+    /**
+     * Inner class to represent the special cell required by the A* algorithm
+     */
+    private class Node {
+        private Node parent;
+        private Position position;
+        private int cost;
+        private int heuristic;
+        private int SumCost;
+
+        Node(Position position) {
+            this.position = position;
+        }
+
+        /**
+         * Creates all neighbouring Nodes of the current Node.
+         * @return a list of Nodes
+         */
+        List<Node> getNeighbours() {
+            List<Node> nodes = new ArrayList<>();
+            nodes.add(new Node(this.getPosition().stepNorth()));
+            nodes.add(new Node(this.getPosition().stepNorthEast()));
+            nodes.add(new Node(this.getPosition().stepEast()));
+            nodes.add(new Node(this.getPosition().stepSouthEast()));
+            nodes.add(new Node(this.getPosition().stepSouth()));
+            nodes.add(new Node(this.getPosition().stepSouthWest()));
+            nodes.add(new Node(this.getPosition().stepWest()));
+            nodes.add(new Node(this.getPosition().stepNorthWest()));
+            return nodes;
+        }
+
+        Node getParent() {
+            return parent;
+        }
+
+        void setParent(Node parent) {
+            this.parent = parent;
+        }
+
+        Position getPosition() {
+            return position;
+        }
+
+        int getCost() {
+            return cost;
+        }
+
+        void setCost(int cost) {
+            this.cost = cost;
+        }
+
+        int getHeuristic() {
+            return heuristic;
+        }
+
+        void setHeuristic(int heuristic) {
+            this.heuristic = heuristic;
+        }
+
+        int getSumCost() {
+            return SumCost;
+        }
+
+        void setSumCost(int sumCost) {
+            SumCost = sumCost;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            Node node = (Node) o;
+
+            return getPosition() != null ? getPosition().equals(node.getPosition())
+                    : node.getPosition() == null;
+
+        }
+
+        @Override
+        public int hashCode() {
+            int result = getParent() != null ? getParent().hashCode() : 0;
```

Comment:
If the `equals` method does not contain the `parent` of the node then it should be removed from the `hashCode` too. I hope that this would not break any existing functionality.